conans/client/command.py

```diff
@@ -58,6 +58,11 @@ def __call__(self, parser, namespace, values, option_strings=None):  # @UnusedVa
                 dest.extend(values)
             except ValueError:
                 dest.append(values)
+        # --build --build=foo == ["*", "foo"]
+        elif hasattr(namespace, "build"):
```

Comment:
Without it, we won't be able to know if `--build` has been passed together with another `--build=foo` or `--build=!foo`. Argparse excludes None entries. As '*' is considered build all from sources and `--build` has the same usage, I don't think it will break Conan contract.